Source/WebKit/UIProcess/API/glib/WebKitWebsiteDataManager.cpp

```diff
@@ -90,7 +90,9 @@ enum {
     PROP_SERVICE_WORKER_REGISTRATIONS_DIRECTORY,
     PROP_DOM_CACHE_DIRECTORY,
 #endif
-    PROP_IS_EPHEMERAL
+    PROP_IS_EPHEMERAL,
+    PROP_PER_ORIGIN_STORAGE_QUOTA,
+    PROP_ORIGIN_STORAGE_RATIO
```

Comment:
There are:
 * totalQuotaRatio
 * originStorageRatio
 * perOriginStorageQuota

originStorageRatio(not specified by default) if is specified it is used instead of perOriginStorageQuota(which is by default 10GB).

totalQuotaRatio is used for whole.

Reply:
And currently there is no API to control totalQuotaRatio

Reply:
Should we add also API for totalQuotaRatio?

Reply:
I suppose. But maybe we'd better talk to @szewai to make sure all of these quotas are likely to remain supported in the future.

Reply:
There are two types of quota now: 
1. Origin quota -- how much can an origin store; if it is exceeded, request will fail.
2. Total quota -- how much can a dataStore (all origins) store; if it is exceeded, eviction will happen.

Origin quota can be configured in two ways:
- Setting value directly -- `perOriginStorageQuota`. This is the existing way of configuring quota.
- Calculating value based on hard disk size -- `originQuotaRatio`: quota = originQuotaRatio * totalDiskSpace. If `originQuotaRatio` is set, calculation will be based on ratio, no matter perOriginStorageQuota is set or not.

Total quota can be configured in one way:
- Calculating value on hard disk size -- `totalQuotaRatio`: quota = totalQuotaRatio * totalDiskSpace. If `totalQuotaRatio` is not set, there is no eviction based on usage.

The motivation for new quota mechanism is to be compatible with other browser engines (https://developer.mozilla.org/en-US/docs/Web/API/Storage_API/Storage_quotas_and_eviction_criteria#how_much_data_can_be_stored). 

I just enabled "calculating quotas based on total disk space" on Cocoa platforms: https://bugs.webkit.org/show_bug.cgi?id=255095. I think we might deprecate `perOriginStorageQuota` if it's not in use anywhere.

Reply:
> I think we might deprecate perOriginStorageQuota if it's not in use anywhere.

Well we don't want to expose it as public API then. Good thing I CCed you.

Reply:
@mcatanzaro: does it mean that this patch should be closed without merging

Reply:
Or maybe just add in this PR API for totalQuotaRatio(and remove: perOriginStorageQuota and originQuotaRatio)?

Reply:
If totalQuotaRatio is all you need, then I would add just that one. Let's expose only what you actually need.

Reply:
Currently the most important from our point of view is perOriginStorageQuota

Reply:
@mcatanzaro : what can we do with this PR?

Reply:
ok, so I removed API for perOriginStorageQuota and added API for totalQuotaRatio